pkgs/by-name/pr/proto/package.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, darwin
+, libiconv
+, makeBinaryWrapper
+, pkg-config
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "proto";
+  version = "0.32.2";
+
+  src = fetchFromGitHub {
+    owner = "moonrepo";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-QxaVXggVfr3O26RwSuUx4qpabhWE9+ZFLKRin/ya2e4=";
+  };
+
+  cargoHash = "sha256-j5wXmlauhrUnUVTj/KyqDrf1y9bq3ZInREZXIUQImVE=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.SystemConfiguration
+    libiconv
+  ];
+  nativeBuildInputs = [ makeBinaryWrapper pkg-config ];
+
+  # Tests requires network access
+  doCheck = false;
+  cargoBuildFlags = [ "--bin proto" "--bin proto-shim" ];
+
+  postInstall = ''
+    # proto looks up a proto-shim executable file in $PROTO_LOOKUP_DIR
+    wrapProgram $out/bin/${pname} \
+      --set PROTO_LOOKUP_DIR $out/bin
+  '';
+
+  meta = {
+    description = "A pluggable multi-language version manager";
+    longDescription = ''
+      proto is a pluggable next-generation version manager for multiple programming languages. A unified toolchain.
+      If you use language runtimes installed by proto, you need to set `''${PROTO_HOME:-$HOME/.proto}/bin` to `PATH` in your shell startup files.
```

Comment:
I'm unsure if this shouldn't rather be a comment in the package. If this is mentioned somewhere in proto's own documentation, we should drop it here. If not, and if it's not NixOS-specific, I'm fine with having it.

Reply:
> If this is mentioned somewhere in proto's own documentation, we should drop it here.

Indeed this is documented in https://moonrepo.dev/docs/proto, so dropped the last line. https://github.com/NixOS/nixpkgs/pull/301492/commits/b42ac809231651028ac50ffb38d41857348b9ee1